packages/titus-backend-typescript/lib/config/index.ts

```diff
@@ -0,0 +1,150 @@
+import path from 'path'
+
+import envSchema from 'env-schema'
+import S from 'fluent-json-schema'
+
+const envJsonSchema = S.object()
+  .prop('NODE_ENV', S.string().required())
+  .prop('API_HOST', S.string().required())
+  .prop('API_PORT', S.string().required())
+  .prop('CORS_ORIGIN', S.string())
+  .prop('ENABLE_ADMIN', S.boolean())
+  .prop('PG_HOST', S.string().required())
+  .prop('PG_PORT', S.string().required())
+  .prop('PG_DB', S.string().required())
+  .prop('PG_USER', S.string().required())
+  .prop('AUTH_PROVIDER', S.string())
+  .prop('AUTH0_DOMAIN', S.string())
+  .prop('AUTH0_CLIENT_ID', S.string())
+  .prop('AUTH0_CLIENT_SECRET', S.string())
+  .prop('AUTH0_AUDIENCE', S.string())
+  .prop('AUTH0_GRANT_TYPE', S.string())
+  .prop('AUTH0_CONNECTION', S.string())
+  .prop('COGNITO_REGION', S.string())
+  .prop('COGNITO_USER_POOL_ID', S.string())
+  .prop('JWT_SECRET', S.string().default('3000'))
+  .prop('AD_TENANT', S.string())
+  .prop('AD_APP_ID', S.string())
+  .prop('AD_SECRET', S.string())
+  .prop('SECRETS_STRATEGY', S.string())
+  .prop('SECRETS_PG_PASS', S.string().required())
+  .prop('HEALTHCHECK_URL', S.string().default('/healthcheck'))
+  .prop('HEALTHCHECK_MAX_HEAP_USER', S.number().default(768 * 1024 * 1024)) // arbitrary, 768 MB of RAM
+  .prop('HEALTHCHECK_MAX_RSS', S.number().default(1024 * 1024 * 1024)) // arbitrary, 1 GB of RAM
+  .prop('HEALTHCHECK_MAX_EVENT_LOOP_UTILIZATION', S.number().default(0.98))
+
+interface IEnv {
```

Comment:
if we would use a tool like https://github.com/sinclairzx81/typebox for the schema definition, we wouldn't have to define the schema second time here. I have tried this, but I bumped into problems when compiling the json schema with ajv.